### ISSUE TITLE
Eq should evaluate when properties are incompatible

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#10682]
e.g.

``` python
>>> Eq(p,var('x',finite=False))
Eq(p, x)
```

Some properties (for which `(lhs-rhs).is_zero` evaluates, where lhs and rhs are the sides of the Eq/Ne) already work:

```
>>> p = Symbol('p', positive=True)
>>> Ne(p, 0)
True  # formerly reported as Ne(p, 0)
```

Any properties which are mutually exclusive should cause Eq/Ne to evaluate. Perhaps this should be part of simplification, not evaluation. 

Questions:
- [ ] how to get all assumptions to evaluate
- [ ] which of `_assumptions`, `_assumptions0` to use

**Comment by sanyagithub** [comment]
>>> p = Symbol('p', positive=True)
>>> Ne(p, 0)
True
>>> Eq(p,0)
False
>>> Eq(p,4)
Eq(p, 4)

I just want to clarify. 
Do you want to say that Eq(p,4) should evaluate to True??

**Comment by thisch** [comment]
No Eq(p,4) should not evaluate to True, since we only know that p is positive and not that p is a concrete Integer. Therefore it makes sense that it does not evaluate. See https://github.com/sympy/sympy/pull/10644#issuecomment-188614649

**Comment by mvnnn** [comment]
@smichr That's issue fixed in PR #10644 so close this issue.
